pkg/providers/cf/provider.go

```diff
@@ -0,0 +1,363 @@
+				}
+			} else {
+				apps = append(apps, app)
+			}
+		}
+	}
+	return apps, nil
+}
+
+// func writeManifest(writer io.Writer, b []byte) error {
+// 	ma := cfTypes.AppManifest{}
+// 	err := yaml.Unmarshal(b, &ma)
+// 	if err != nil {
+// 		return err
+// 	}
+// 	a, err := Discover(ma)
+// 	if err != nil {
+// 		return err
+
+// 	}
+// 	fmt.Printf("discovered manifest: %v\n", a)
+// 	b, err = yaml.Marshal(a)
+// 	if err != nil {
+// 		return err
+
+// 	}
+// 	if output == "" {
+// 		fmt.Fprintf(writer, "%s", b)
+// 		return nil
+// 	}
+// 	return os.WriteFile(output, b, 0644)
+// }
+
+func createApp(cfApp cfTypes.AppManifest) (dTypes.Application, error) {
+	timeout := 60
+	if cfApp.Timeout != 0 {
+		timeout = int(cfApp.Timeout)
+	}
+	services, err := p.MarshalUnmarshal[dTypes.Services](cfApp.Services)
+	if err != nil {
+		return dTypes.Application{}, err
+	}
+	routeSpec := p.ParseRouteSpec(cfApp.Routes, cfApp.RandomRoute, cfApp.NoRoute)
+	docker, err := p.MarshalUnmarshal[dTypes.Docker](cfApp.Docker)
+
+	if err != nil {
+		return dTypes.Application{}, err
+	}
+	sidecars, err := p.MarshalUnmarshal[dTypes.Sidecars](cfApp.Sidecars)
+	if err != nil {
+		return dTypes.Application{}, err
+	}
+	processes, err := p.MarshalUnmarshal[dTypes.Processes](cfApp.Processes)
+	if err != nil {
+		return dTypes.Application{}, err
+	}
+	var labels, annotations map[string]*string
+
+	if cfApp.Metadata != nil {
+		labels = cfApp.Metadata.Labels
+		annotations = cfApp.Metadata.Annotations
+	}
+	appManifestProcess, inlineProcess, err := p.ParseProcessSpecs(cfApp)
+
+	if err != nil {
+		return dTypes.Application{}, err
+	}
+	var appManifestProcessTemplate *dTypes.ProcessSpecTemplate
+
+	if appManifestProcess != nil {
+		appManifestProcessTemplate = appManifestProcess
+	}
+	if inlineProcess != (nil) {
+		processes = append(processes, *inlineProcess)
+	}
+	app := dTypes.Application{
+		Metadata: dTypes.Metadata{
+			Name:        cfApp.Name,
+			Labels:      labels,
+			Annotations: annotations,
+		},
+		Timeout:    timeout,
+		BuildPacks: cfApp.Buildpacks,
+		Env:        cfApp.Env,
+		Stack:      cfApp.Stack,
+		Services:   services,
+		Routes:     routeSpec,
+		Docker:     docker,
+		Sidecars:   sidecars,
+		Processes:  processes,
+	}
+	if appManifestProcessTemplate != nil {
+		app.ProcessSpecTemplate = *appManifestProcessTemplate
+	}
+	validationErrors := validateApplication(app)
+	if validationErrors != nil {
+		return dTypes.Application{}, errors.Join(validationErrors...)
+	}
+	return app, nil
+}
+
+func (c *CFProvider) createCFManifest(spaceName string) ([]cfTypes.AppManifest, error) {
+	ctx := context.Background()
+	log.Println("Analyzing space: ", spaceName)
+	spaceOpts := client.NewSpaceListOptions()
+	spaceOpts.Names.EqualTo(spaceName)
+	cfClient, err := c.GetClient()
+	if err != nil {
+		return nil, fmt.Errorf("error creating CF client: %v", err)
+	}
+	remoteSpace, err := cfClient.Spaces.First(ctx, spaceOpts)
```

Comment:
what do you think to extract this few lines to a function that given a space name, returns the list of apps in that space?